types/module/module.go

```diff
@@ -383,27 +383,33 @@ func (m *Manager) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) map[string
 // ExportGenesisForModules performs export genesis functionality for modules
 func (m *Manager) ExportGenesisForModules(ctx sdk.Context, cdc codec.JSONCodec, modulesToExport []string) map[string]json.RawMessage {
 	genesisData := make(map[string]json.RawMessage)
-	if len(modulesToExport) == 0 {
-		for _, moduleName := range m.OrderExportGenesis {
-			if module, ok := m.Modules[moduleName].(HasGenesis); ok {
-				genesisData[moduleName] = module.ExportGenesis(ctx, cdc)
-			}
-		}
 
-		return genesisData
+	if len(modulesToExport) == 0 {
+		modulesToExport = m.OrderExportGenesis
 	}
 
 	// verify modules exists in app, so that we don't panic in the middle of an export
 	if err := m.checkModulesExists(modulesToExport); err != nil {
 		panic(err)
 	}
 
+	var channels []chan json.RawMessage
```

Comment:
Just curious, it isn't better to assign the size of the channel slice here at creation of the slice.
And assign and create the channel later in the loop?

Feel free to dismiss if you think not.

like:

```suggestion
	channels := make([]chan json.RawMessage, len(modulesToExport))
```